### ISSUE TITLE
Only presets are reloaded with the binding, selected one isn't applied

### ISSUE THREAD

**Issue by BlueGuitarCL** [Mrcubix/Preset.Binding#2]
> Please read this alongside the [LIVE demo](https://youtu.be/YR72cWv61Vo?t=16700s) until 4:44:54 mark, where I have the `otd-daemon` output and the UI open side by side.

Title is self explanatory. But here are some things to note:
- My setup is a CTL-4100WL using **OTL v0.6.4.0** on **ArchLinux (kernel used: `6.11.3-zen**1-1-zen`).
- The configuration shown are three presets bound each one to an auxiliary button of the Wacom Intuos BT, on the demo it can be seen that **on all presets the bindings are the same**, `UX Remote` check mark enabled, and the only difference is having different screen/tablet area mappings per preset.
- **When enabled manually, each preset works as expected** (moving the cursor inside the defined area).

**Comment by Mrcubix** [comment]
[Preset.Binding.zip](https://github.com/user-attachments/files/17719264/Preset.Binding.zip)

![image](https://github.com/user-attachments/assets/cb1c9317-65f8-4e87-bc9e-6a4741df17f0)

This build only adds a try catch in case it's an exception happening on my side.

This may be able to provide me with more info, as i literally cannot reproduce this on my Windows, Mint & Raspi OS installs (which make me think this is a Arch-only issue)

Also, please attach a diagnostic file (Help > Export diagnostics in the UX)

![image](https://github.com/user-attachments/assets/6b615b2a-f9d7-4f13-84ab-93da2df1d74c)

Also if switching preset fails, it's supposed to return an error message, and it isn't the case on your side, so either it's hanging forever (unlikely) or it succeeded

**Comment by Mrcubix** [comment]
You may try this build : https://github.com/Mrcubix/Preset.Binding/actions/runs/12129991732/artifacts/2264526952

In the case where either the UX remote isn't connected, it will fallback to built-in notifications.
Otherwise, if the UX remote is connected, it will synchronize, hopefully send the notification, and if any of those 2 operations fail, it shouldn't kill inputs (it usually does if an unhandled exception occur.

**Comment by BlueGuitarCL** [comment]
> [Preset.Binding.zip](https://github.com/user-attachments/files/17719264/Preset.Binding.zip)

Using this build, i got this error for all my preset binds.
```
[OTD Presets:Error]	Error: System.InvalidOperationException: This operation is not allowed before listening for messages has started.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync(JsonRpcRequest request, Type expectedResultType, CancellationToken cancellationToken)
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at OTD.PresetBinds.IPC.Remote.ApplySettingsAsync(Settings settings, String name)
```

> Also, please attach a diagnostic file (Help > Export diagnostics in the UX)

Here's the diagnostics JSON file of that one.
[otd-preset-bindings-issue2-comment_build.json](https://github.com/user-attachments/files/17985168/otd-preset-bindings-issue2-comment_build.json)

I'll download CI build n°4 and test that too. I'll reply back my findings on a short while.

**Comment by BlueGuitarCL** [comment]
![image](https://github.com/user-attachments/assets/c30d3eee-6b2f-435f-b7be-727c35bf47f3)
Btw, with CI build n°6 `otd-daemon` starts, but as soon as i launch the GUI, the daemon floods with errors and doesn't let the GUI initialize

**Comment by Mrcubix** [comment]
Can't check your screenshot in details cause github is griefing, need to upload it elsewhere

**Comment by Mrcubix** [comment]
Also, here is a clip of me trying the changes on my raspberry pi with a wireless wacom tablet, while VSCode is running in the background (Absolute worst case scenario) 

https://youtu.be/k091kaQ7ipI

[Preset.Binding.zip](https://github.com/user-attachments/files/17985934/Preset.Binding.zip)
This build contains debug symbols, maybe you'll be able to get more details out of those exceptions?

But anyway, if you don't need UI feedback on switches, you can remove UX Remote, it's purely there to synchronize the UX, & send notification with a proper header.

**Comment by Mrcubix** [comment]
Ok so it's possible this is due to OTD's old plugin memory leak.
Short explanation is : OTD doesn't dispose of unused plugin instances when the device is disconnected.

The result of that poor design is instances still existing, and being abandoned.
So plug your devices in enough time or do other specific things enough time and you're bound to meet a race condition.

I can't fix that, but i can at least make sure that I'm not already attempting a connection.

[Preset.Binding.zip](https://github.com/user-attachments/files/17986585/Preset.Binding.zip)

**Comment by BlueGuitarCL** [comment]
Alright, here are the 2° comment build results:
[otd-preset-bindings-issue2-comment_build_2-w_video.json](https://github.com/user-attachments/files/17986694/otd-preset-bindings-issue2-comment_build_2-w_video.json)
https://youtu.be/uDDfb7HRg44

Not sure if this is relevant, but note on 0:55 this popped on the log when i clicked "Pen Settings".
```
     {
      "Time": "2024-12-02T22:53:16.6481346-03:00",
      "Group": "AggregateException",
      "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (This operation is not allowed after listening for messages has started.)",
      "StackTrace": null,
      "Level": 3,
      "Notification": false
    },
    {
      "Time": "2024-12-02T22:53:16.6502075-03:00",
      "Group": "IPC",
      "Message": "Disconnected from a client.",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    }
```
And here is the relevant part, the thing that prompted on the 2nd try.

```
[Settings:Info]	Loaded preset 'ctl4100wl-artist-eDP1.json'
[Settings:Info]	Loaded preset 'ctl4100wl-artist-full.json'
[Settings:Info]	Loaded preset 'ctl4100wl-artist-HDMI1.json'
[Settings:Info]	Presets have been refreshed. Loaded 3 presets.
[Wacom CTL-4100WL:Info]	Output mode: Artist Mode
[Wacom CTL-4100WL:Info]	Display area: [1366x768@<960, 1464>:0°],
[Wacom CTL-4100WL:Info]	Tablet area: [152x85.45827@<76, 42.729134>:0°],
[Wacom CTL-4100WL:Info]	Clipping: Enabled
[Wacom CTL-4100WL:Info]	Ignoring reports outside area: Disabled
[Wacom CTL-4100WL:Info]	Tip Binding: [OpenTabletDriver.Desktop.Binding.LinuxArtistMode.LinuxArtistModeButtonBinding]@0%
[Wacom CTL-4100WL:Info]	Pen Bindings: OpenTabletDriver.Desktop.Binding.LinuxArtistMode.LinuxArtistModeButtonBinding, OpenTabletDriver.Desktop.Binding.LinuxArtistMode.LinuxArtistModeButtonBinding
[Wacom CTL-4100WL:Info]	Express Key Bindings: Multi-Key Binding: Control+Z, OTD.PresetBinds.Binding.PresetBinding, OTD.PresetBinds.Binding.PresetBinding, OTD.PresetBinds.Binding.PresetBinding
[Settings:Info]	Driver is enabled.
[Tool:Error]	Failed to initialize  tool.
[OTD Presets:Debug]	Switched to 'ctl4100wl-artist-eDP1' preset
[OTD Presets:Info]	Switched to 'ctl4100wl-artist-eDP1' preset
[NullReferenceException:Error]	Object reference not set to an instance of an object.
   at OpenTabletDriver.Plugin.Output.AbsoluteOutputMode.OnOutput(IDeviceReport report) in /home/main-builder/pkgwork/src/OpenTabletDriver-0.6.4.0/OpenTabletDriver.Plugin/Output/AbsoluteOutputMode.cs:line 167
   at OpenTabletDriver.InputDeviceTree.HandleReport(Object sender, IDeviceReport report) in /home/main-builder/pkgwork/src/OpenTabletDriver-0.6.4.0/OpenTabletDriver/InputDeviceTree.cs:line 65
   at OpenTabletDriver.Devices.DeviceReader`1.Main() in /home/main-builder/pkgwork/src/OpenTabletDriver-0.6.4.0/OpenTabletDriver/Devices/DeviceReader.cs:line 111
[AggregateException:Error]	A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (An internal error occurred. Please contact Microsoft Support.)
```

It's strange that it did work only once without an error. I'll try now with the build you uploaded like ~30m ago.

**Comment by BlueGuitarCL** [comment]
And for the 3° comment build... it seems to do nothing, no changes or errors after using the shortcuts.
Clip: https://youtu.be/3iZLW2TGmdA
Log: [otd-preset-bindings-issue2-comment_build_3-w_video.json](https://github.com/user-attachments/files/17986990/otd-preset-bindings-issue2-comment_build_3-w_video.json)

Edit: I didn't try switching to ...`-eDP1` preset first on this video, but doing the same procedure as the previous clip also throwed the same "NullReferenceException".
So just in case, I'll create 2 brand new presets also on artist mode, each one binded to one screen and with only one "Preset Binding". The remaining 3 buttons will have no binds.

**Comment by Mrcubix** [comment]
Good catch on the absolute mode exception, it's an issue OTD was trying to track for a long time (but out of scope here)

Edit: unfortunately, the stacktrace contains the wrong line, so would have to null check all to guess which one causes the issue.

Could narrow it down if you debug the daemon. (You may also build and remote debug my plugin using the vscode launch config if you want to make this plugin issue easier to identify) (you might need to remove the exe extension in my launch config)

As for the logs with the plugin, could you provide the ones from the services log instead? Might contain more complete stacktraces

**Comment by BlueGuitarCL** [comment]
> As for the logs with the plugin, could you provide the ones from the services log instead?
> Might contain more complete stacktraces

Here's systemd logs as verbose as you can get.
```
journalctl _SYSTEMD_INVOCATION_ID=`systemctl --user show --value -p InvocationID opentabletdriver.service` -p 7 -o verbose --no-pager > otd-preset-bindings-issue2-comment_build_3-journalctl_-p_7_-o_verbose.log
```
[otd-preset-bindings-issue2-comment_build_3-journalctl_-p_7_-o_verbose.log](https://github.com/user-attachments/files/17987261/otd-preset-bindings-issue2-comment_build_3-journalctl_-p_7_-o_verbose.log)

But you can see it won't show nothing new if the service/executable (here OTD) is still up and running.

> Could narrow it down if you debug the daemon. (You may also build and remote debug my plugin using the vscode launch config if you want to make this plugin issue easier to identify) (you might need to remove the exe extension in my launch config)

Could you describe exactly what would be the procedure to "debug the daemon"? I don't frequently use VSCode, so no idea how to prep the Remote Debug/Launch Config stuff tbh.
Also, i'm not sure if i'll keep myself awake around now. It's 1:31am here.

**Comment by Mrcubix** [comment]
No problem, sleep is important to most, i'm just not part of those.

It's almost 6 AM here, but basically:

You'll need the .NET 6 SDK (which may not be easily installable nowadays)

- for my plugin : 

1. change the build config in build.sh to `Debug` from `Release` (-c arg).
Run the build script (it might fail at the archuve creation part with jar).
2. Copy the dll & pdb file generated in the `build` directory to Preset.Binding plugin Folder (Overwriting where necessary)
3. Launch the daemon or service
4. Start debugging in vscode (Default option should be "OpenTabletDriver C# Plugin Debugging")
5. Any exception within the plugin should pause at the exception.

(If that's not the case, switch to the bug icon on the left, which is the debugging tab, and in `breakpoints` tick both handled & unhandled exception)

As for OpenTabletDriver, you may either launch it in your IDE of choice or build a debug build of it:

1. Cd to the root of OpenTabletDriver
2. Use `./eng/Linux/package.sh` (or something simillar) and hopefully, debug binaries will be generated in the `dist` directory, also at the root. You may use those.

To remote debug that build, you can just debug like the plugin (it remote debug the daemon)

**Comment by BlueGuitarCL** [comment]
- I managed to build [OTD v0.6.4.0 Source](https://github.com/OpenTabletDriver/OpenTabletDriver/releases/tag/v0.6.4.0) without issues.
- After obtaining [Preset.Binding Release 1.1.1](https://github.com/Mrcubix/Preset.Binding/releases/tag/1.1.1) (not using latest commit, bc that is the CI build °6 that [i've already tested](https://github.com/Mrcubix/Preset.Binding/issues/2#issuecomment-2513233763))
- ... populating its `.modules` directory with  `git clone --depth 1 --branch 0.6.x --single-branch git@github.com:Mrcubix/OpenTabletDriver.External.git` and a symlink to OTD's Source
- ... and populating OTD.External's `.modules` dir again with OTD's Source
greeted me with this:
```
$ ./build.sh
MSBuild version 17.8.5+b5265ef37 for .NET
  Determining projects to restore...
/home/blu/Sources/Preset.Binding-1.1.1/.modules/OTD.UX.Remote/OTD.UX.Remote.Lib/OTD.UX.Remote.Lib.csproj : error NU1201: Project OpenTabletDriver.External.Common is not compatible with netstandard2.1 (.NETStandard,Version=v2.1). Project OpenTabletDriver.External.Common supports: net6.0 (.NETCoreApp,Version=v6.0) [/home/blu/Sources/Preset.Binding-1.1.1/Preset.Binding.sln]
  Failed to restore /home/blu/Sources/Preset.Binding-1.1.1/.modules/OTD.UX.Remote/OTD.UX.Remote.Lib/OTD.UX.Remote.Lib.csproj (in 352 ms).
  Restored /home/blu/Sources/Preset.Binding-1.1.1/src/Preset.Binding.csproj (in 468 ms).
  Restored /home/blu/Sources/Preset.Binding-1.1.1/.modules/OTD.UX.Remote/.modules/OpenTabletDriver.External/.modules/OpenTabletDriver/OpenTabletDriver.Desktop/OpenTabletDriver.Desktop.csproj (in 394 ms).
  Restored /home/blu/Sources/Preset.Binding-1.1.1/.modules/OTD.UX.Remote/.modules/OpenTabletDriver.External/OpenTabletDriver.External.Common/OpenTabletDriver.External.Common.csproj (in 749 ms).
  5 of 9 projects are up-to-date for restore.
[...]
```
So i haven't performed any tests or debugging since i dont have `Preset.Binding` built correctly on my system yet.

3:20am already. I'll call the night here. Let me know if you got some findings regarding the issue, or advice with the .NET tooling that i'm not used to.

**Comment by Mrcubix** [comment]
`git submodule update --init --recursive` should had gotten you the correct result submodule wise.

You're getting this error cause you either fetched the wrong OpenTabletDriver.External Branch (Should be master) 
(you can choose to only fetch the OpenTabletDriver.Common project there)

This should should allow you to build the plugin.

**Comment by BlueGuitarCL** [comment]
This took me longer than what i've expected

> `git submodule update --init --recursive` should had gotten you the correct result submodule wise.

Taking note your suggestion, i did this instead:
```
git clone git@github.com:Mrcubix/Preset.Binding.git; cd Preset.Binding
git checkout tags/1.1.1
git submodule update --init --recursive
chmod +x build.sh; ./build.sh
```

And now the plugin + the daemon built succesfully. (Aside of `SharpZipLib` vulnerability warnings, and `cp: cannot stat 'temp/Preset.Binding.pdb': No such file or directory`)

When debugging time came up, i got a prompt saying something like "`coreclr` not supported". Turns out that `vsdbg` [isn't licensed for use with e.g. VSCodium](https://github.com/dotnet/vscode-csharp/issues/1431#issuecomment-297578930), so i had to use `netcoredbg` provided by an extension instead. (`@category:debuggers coreclr` on ext. marketplace)

Nothing seems to happen without breakpoints, but got a hit enabling the `all` breakpoint checkmark:
![image](https://github.com/user-attachments/assets/5d1945c0-2273-457f-8c53-ce2bf197d28b)
```
[NullReferenceException:Error]	Object reference not set to an instance of an object.
   at OpenTabletDriver.Plugin.Output.AbsoluteOutputMode.OnOutput(IDeviceReport report) in /home/blu/Sources/OpenTabletDriver-0.6.4.0/OpenTabletDriver.Plugin/Output/AbsoluteOutputMode.cs:line 156
   at OpenTabletDriver.Plugin.Output.OutputMode.Consume(IDeviceReport report) in /home/blu/Sources/OpenTabletDriver-0.6.4.0/OpenTabletDriver.Plugin/Output/OutputMode.cs:line 112
   at OpenTabletDriver.Plugin.Output.OutputMode.Read(IDeviceReport deviceReport) in /home/blu/Sources/OpenTabletDriver-0.6.4.0/OpenTabletDriver.Plugin/Output/OutputMode.cs:line 115
   at OpenTabletDriver.InputDeviceTree.HandleReport(Object sender, IDeviceReport report) in /home/blu/Sources/OpenTabletDriver-0.6.4.0/OpenTabletDriver/InputDeviceTree.cs:line 63
   at OpenTabletDriver.Devices.DeviceReader`1.OnReport(T report) in /home/blu/Sources/OpenTabletDriver-0.6.4.0/OpenTabletDriver/Devices/DeviceReader.cs:line 140
   at OpenTabletDriver.Devices.DeviceReader`1.Main() in /home/blu/Sources/OpenTabletDriver-0.6.4.0/OpenTabletDriver/Devices/DeviceReader.cs:line 111
   ```
After that i get several pauses on exceptions from 3 of 6 of the .NET ThreadPool Workers i've got, and (if i disable all breakpoints,) the tablet is unresponsive after the `NullReferenceException` we've been discussing.

Edit: Here's something regarding the StreamJsonRpc errors we had earlier:
![image](https://github.com/user-attachments/assets/4d484b4d-1e80-4239-89f1-11e368f36d1e)

**Comment by Mrcubix** [comment]
> After that i get several pauses on exceptions from 3 of 6 of the .NET ThreadPool Workers i've got, and (if i disable all breakpoints,) the tablet is unresponsive after the NullReferenceException we've been discussing.

Can you check, which part specifically is null (it's prob on the tablet spec part) ? You may also add ? after each parts like so temporarily:

```cs
pressureHandler.SetPressure(tabletReport.Pressure / (float)(Tablet?.Properties?.Specifications?.Pen?.MaxPressure ?? 8191));
```

> Here's something regarding the StreamJsonRpc errors we had earlier:

That is indeed very odd, means that either `Driver` is not connected (which is unlikely) or that it's not attached yet (why would it not be attached at that point)

Checking if it's connected first wouldn't make sense, because i already check if it's connected in `PresetBinding.Press()`.

[~~OpenTabletDriver.External.patch~~](https://github.com/user-attachments/files/17996838/OpenTabletDriver.External.patch)
[~~Preset.Binding.patch~~](https://github.com/user-attachments/files/17996890/Preset.Binding.patch)
[~~Remote.patch~~](https://github.com/user-attachments/files/17996891/Remote.patch)

~~Could you try applying these 3 patches and see if that fixes it, otherwise, i can push my changes to be tested.~~

**Comment by BlueGuitarCL** [comment]
I won't be available for around ~4 hours. I'll test the patches as soon as i have time again.

**Comment by Mrcubix** [comment]
You may just pull my changes, made sure to update the dependencies.

If you made any changes you might need to fetch then reset to new master (or discard your changes in the Source Control tab of vscode)

**Comment by BlueGuitarCL** [comment]
New clip here: https://youtu.be/VjvQy4xobMc
Seems like even a Multi-Key Bind on an Auxiliary button isn't working now that i've built commit `3bf9b73`. Also, i'll be on the same situation as before if i re-enable the `all` exception breaklpoints, because those that occurs on the `.NET ThreadPool Workers` have no error descriptions associated.

Can you provide steps to build all the DLL's listed here with debug symbols? The debugger shows this on red on the clip at the daemon startup. I think without this we're debugging blindfolded.
```
You are debugging a Release build of ...
OpenTabletDriver.Daemon.dll
OpenTabletDriver.dll
OpenTabletDriver.Desktop.dll
OpenTabletDriver.Plugin.dll
Octokit.dll
OpenTabletDriver.Configurations.dll
OpenTabletDriver.Native.dll
Preset.Binding.dll
. Using Just My Code with Release builds using compiler optimizations results in a degraded debugging experience (e.g. breakpoints will not be hit).
```

I wonder if you can reproduce the issue on the ArchLinux ~ISO without installing~ VM (i mean the x86_64 distro, bc i'm not familiar with the ARM one). I'll try setting that up meanwhile.

**Comment by Mrcubix** [comment]
> I wonder if you can reproduce the issue on the ArchLinux ISO VM

I need to setup another VM that isn't just debian or Ubuntu based eventually, not now however.

> Can you provide steps to build all the DLL's listed here with debug symbols?

You don't have to worry about those (Except for the plugin) when debugging from the IDE.

```diff
diff --git a/.vscode/launch.json b/.vscode/launch.json
index ef0cbb09..51b62e01 100644
--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -8,6 +8,7 @@
             "preLaunchTask": "Build Daemon",
             "cwd": "${workspaceFolder}/OpenTabletDriver.Daemon/bin/Debug/net8.0",
             "program": "${workspaceFolder}/OpenTabletDriver.Daemon/bin/Debug/net8.0/OpenTabletDriver.Daemon.dll",
+            "justMyCode": false,
             "args": [
```

This should allow you to debug both OTD & the plugin (provided it's built with debug configuration)

## Building the plugin

Edit build.sh, change configuration from `Release` to `Debug`

E.G: `-c Release` -> `-c Debug`

**Comment by Mrcubix** [comment]
IF you really want a separate build of OTD, use these :

![image](https://github.com/user-attachments/assets/58369f40-e8a5-4c32-b21a-52b84fde5410)

Change `CONFIG` to Debug

then from the root, run `./eng/linux/package.sh`

This should generate a debug build in the dist directory as specified before (this time in debug since i forgot that step last time)

**Comment by BlueGuitarCL** [comment]
I decided to download the Microsoft build of VSCode so i can try using `vsdbg`, and that solves the random `.NET ThreadPool Workers` exceptions i had when debugging with `netcoredbg` ¯\\\_(ツ)\_/¯

> IF you really want a separate build of OTD, use these :

Yeah... i finished building OTD entirely to make sure the repo works as expected. I'll list again the current procedure:
```
git clone git@github.com:Mrcubix/Preset.Binding.git; cd Preset.Binding
git submodule update --init --recursive; (change everything to Debug builds. The only one left is Octokit.dll)
./build.sh; .modules/OpenTabletDriver/build.sh
ln -s *.{dll,pdb} ~/.config/OpenTabletDriver/Plugins/Preset.Binding/
```

and doing a slight change to the `launch.json`:
```diff
diff --git a/.vscode/launch.json b/.vscode/launch.json
index 63940ac..c0669d6 100644
--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -4,9 +4,9 @@
         {
             "name": "OpenTabletDriver C# Plugin Debugging",
             "type": "coreclr",
-            "request": "attach",
+            "request": "launch",
             "requireExactSource": false,
-            "processName": "OpenTabletDriver.Daemon",
+            "program": "${workspaceFolder}/.modules/OpenTabletDriver/bin/OpenTabletDriver.Daemon.dll",
             "logging": {
                 "moduleLoad": false
             },
```

> Can you check, which part specifically is null (it's prob on the tablet spec part) ? You may also add ? after each parts like so temporarily:
> 
> ```cs
> pressureHandler.SetPressure(tabletReport.Pressure / (float)(Tablet?.Properties?.Specifications?.Pen?.MaxPressure ?? 8191));
> ```

Tried applying this, but still got `NullPointerException`. Screnshot:
![image](https://github.com/user-attachments/assets/8c387830-cf82-43fc-97c6-f9c5e3735a69)
Call Stack:
```
Exception has occurred: CLR/System.NullReferenceException
Exception thrown: 'System.NullReferenceException' in OpenTabletDriver.Plugin.dll: 'Object reference not set to an instance of an object.'
   at OpenTabletDriver.Plugin.Output.AbsoluteOutputMode.OnOutput(IDeviceReport report) in /home/blu/Sources/Preset.Binding/.modules/OpenTabletDriver/OpenTabletDriver.Plugin/Output/AbsoluteOutputMode.cs:line 145

OpenTabletDriver.Plugin.dll!OpenTabletDriver.Plugin.Output.AbsoluteOutputMode.OnOutput(OpenTabletDriver.Plugin.Tablet.IDeviceReport report) Line 145 (/home/blu/Sources/Preset.Binding/.modules/OpenTabletDriver/OpenTabletDriver.Plugin/Output/AbsoluteOutputMode.cs:145)
OpenTabletDriver.Plugin.dll!OpenTabletDriver.Plugin.Output.OutputMode.Consume(OpenTabletDriver.Plugin.Tablet.IDeviceReport report) Line 113 (/home/blu/Sources/Preset.Binding/.modules/OpenTabletDriver/OpenTabletDriver.Plugin/Output/OutputMode.cs:113)
OpenTabletDriver.Plugin.dll!OpenTabletDriver.Plugin.Output.OutputMode.Read(OpenTabletDriver.Plugin.Tablet.IDeviceReport deviceReport) Line 116 (/home/blu/Sources/Preset.Binding/.modules/OpenTabletDriver/OpenTabletDriver.Plugin/Output/OutputMode.cs:116)
OpenTabletDriver.dll!OpenTabletDriver.InputDeviceTree.HandleReport(object sender, OpenTabletDriver.Plugin.Tablet.IDeviceReport report) Line 60 (/home/blu/Sources/Preset.Binding/.modules/OpenTabletDriver/OpenTabletDriver/InputDeviceTree.cs:60)
OpenTabletDriver.dll!OpenTabletDriver.Devices.DeviceReader<OpenTabletDriver.Plugin.Tablet.IDeviceReport>.OnReport(OpenTabletDriver.Plugin.Tablet.IDeviceReport report) Line 140 (/home/blu/Sources/Preset.Binding/.modules/OpenTabletDriver/OpenTabletDriver/Devices/DeviceReader.cs:140)
OpenTabletDriver.dll!OpenTabletDriver.Devices.DeviceReader<OpenTabletDriver.Plugin.Tablet.IDeviceReport>.Main() Line 111 (/home/blu/Sources/Preset.Binding/.modules/OpenTabletDriver/OpenTabletDriver/Devices/DeviceReader.cs:111)
[External Code] (Unknown Source:0)
```

Also, i think i sorted out a way to get this error reliably on a multi-screen setup.
1. Create a preset that maps the drawing tablet to Screen B
2. Apply a config that maps the drawing tablet to Screen A
3. Keep hovering the pencil to move the cursor over the "File>Presets" dropdown menu and select the presets that changes from current Screen A to Screen B mapping with the pen tip or clicking with a mouse(pad). Above exception occurs.

Yes, i understand this particular thing goes out the scope of the `Preset.Binding` plugin, and would be a good idea mentioning this issue to the aforementioned OTD's gh issue tracker tbh.
Also, note that bc i'm using the minimum version of OTD supported for your plugin, I've been using the daemon + UX.Gtk `v0.6.0.2` (i've just noticed about this). This might be the reason of why the Aux Buttons aren't working. So tests are missing to discard if the unresponsive Aux Buttons are a broken feature somewhere between this ver. and `v0.6.4.0`, or something between `1.1.1` and commit `3bf9b737` of your plugin. Don't know, i'll take some rest for now.

**Comment by Mrcubix** [comment]
> Tried applying this, but still got NullPointerException. Screenshot: 

you can hover on each parts of the of the properties to see which one is null

![image](https://github.com/user-attachments/assets/c56f4688-ee3d-4723-bb1c-00aafdd4671a)

In you case, you can do that for every parts after `Tablet.Specifications` since that's where i expect the null value to be 
(unless it's somehow is in the report, then *what*)

Btw, you should report it to the [Main Repo](https://github.com/OpenTabletDriver/OpenTabletDriver), since you have the stacktrace on top of that, it's a known issue but i don't think we go precise info about it.

The issue is specifically doing it while the pen is in range, and only for some people & tablets (which can cause issues with either the parser or various parts up to the output mode).

> This might be the reason of why the Aux Buttons aren't working

Are you talking about them not working at all, or only with my plugin, if it's the former, then check Tablet Debugger, if it's the later then I've noticed that Aux keys will stop doing anything for a small while after changing preset twice

**Comment by Mrcubix** [comment]
Finally managed to get a arch install working (their installer barely works)

https://github.com/user-attachments/assets/6835e08b-9fcf-4f1d-925e-b0eadc461425

As you can see, I'm unable to replicate the issue, but the daemon hasn't been running for long.

**Comment by BlueGuitarCL** [comment]
I waited until https://github.com/OpenTabletDriver/OpenTabletDriver/pull/3614 got merged and released into v0.6.5.1, and
here's [a vid of me testing again](https://youtu.be/Ji7NdjA-M9M).

Prep stuff i did before recording, to start over:
```
$ git clone git@github.com:Mrcubix/Preset.Binding.git --recurse-submodules --depth 1 --single-branch
$ git diff
diff --git a/build.sh b/build.sh
index 8b7e1c0..c7bd44a 100644
--- a/build.sh
+++ b/build.sh
@@ -2,7 +2,7 @@

 files=( Preset.Binding OpenTabletDriver.External.Common OTD.UX.Remote.Lib )

-dotnet publish -c Release -o temp
+dotnet publish -c Debug -o temp

 # Move the Preset.Binding.dll to the build folder
 # check if the build folder does not exist, else create it
$ chmod +x build.sh; ./build.sh
$ ln -sT ~/Sources/Preset.Binding/build/ ~/.config/OpenTabletDriver/Plugins/Preset.Binding
$ systemctl --user restart opentabletdriver.service
$ journalctl --user-unit=opentabletdriver --follow
```
Because with last commit the bug was fixed, im closing the issue.

P.D.: I wonder if with the recents PR's pushed into OTD v0.6.5.1, your `UX.Remote` plugin is not longer required.
P.D.2: Offtopic, I really wish I had the option as you to get wireless on my tablet working, but it seems im out of luck there because an OTD's HIDSharp dependency not recognizing BT devices well enough (Im reffering to this -> https://github.com/OpenTabletDriver/OpenTabletDriver/pull/2628#issuecomment-1412535654 ). If it wasn't for that, I'd be happy to debug as I did on this issue in order to get wireless working on this device. And I don't think I'm capable of making a diffpatch of OTD's master branch just for wireless to work lol.

**Comment by Mrcubix** [comment]
UX.Remote was made optional shortly after support was added, it only serves to update the UX.
UX.Remote will continue acting after uninstall until OTD is restarted.

As for the wireless part, i'm using Wacom's wireless dongle, which ends up being a USB device.
